lightning_examples/finetuning-scheduler/finetuning-scheduler.py

```diff
@@ -0,0 +1,712 @@
+            text_pairs, max_length=self.max_seq_length, padding="longest", truncation=True
+        )
+        # Rename label to labels to make it easier to pass to model forward
+        features["labels"] = example_batch["label"]
+        return features
+
+
+# %%
+class RteBoolqModule(pl.LightningModule):
+    """A ``LightningModule`` that can be used to finetune a foundational model on either the RTE or BoolQ SuperGLUE
+    tasks using Hugging Face implementations of a given model and the `SuperGLUE Hugging Face dataset."""
+
+    def __init__(
+        self,
+        model_name_or_path: str,
+        optimizer_init: Dict[str, Any],
+        lr_scheduler_init: Dict[str, Any],
+        model_cfg: Optional[Dict[str, Any]] = None,
+        task_name: str = DEFAULT_TASK,
+        experiment_tag: str = "default",
+    ):
+        """
+        Args:
+            model_name_or_path (str): Path to pretrained model or identifier from https://huggingface.co/models
+            optimizer_init (Dict[str, Any]): The desired optimizer configuration.
+            lr_scheduler_init (Dict[str, Any]): The desired learning rate scheduler config
+            model_cfg (Optional[Dict[str, Any]], optional): Defines overrides of the default model config. Defaults to
+                ``None``.
+            task_name (str, optional): The SuperGLUE task to execute, one of ``'rte'``, ``'boolq'``. Defaults to "rte".
+            experiment_tag (str, optional): The tag to use for the experiment and tensorboard logs. Defaults to
+                "default".
+        """
+        super().__init__()
+        self.optimizer_init = optimizer_init
+        self.lr_scheduler_init = lr_scheduler_init
+        if task_name in TASK_NUM_LABELS.keys():
+            self.task_name = task_name
+        else:
+            self.task_name = DEFAULT_TASK
+            rank_zero_warn(f"Invalid task_name '{task_name}'. Proceeding with the default task: '{DEFAULT_TASK}'")
+        self.num_labels = TASK_NUM_LABELS[self.task_name]
+        self.experiment_id = f"{datetime.now().strftime('%Y%m%d_%H%M%S')}_{experiment_tag}"
+        self.model_cfg = model_cfg or {}
+        conf = AutoConfig.from_pretrained(model_name_or_path, num_labels=self.num_labels, local_files_only=False)
+        self.model = AutoModelForSequenceClassification.from_pretrained(model_name_or_path, config=conf)
+        self.model.config.update(self.model_cfg)  # apply model config overrides
+        self.init_hparams = {
+            "optimizer": self.optimizer_init,
+            "lr_scheduler": self.lr_scheduler_init,
+            "model_config": self.model.config,
+            "model_name_or_path": model_name_or_path,
+            "task_name": self.task_name,
+            "experiment_id": self.experiment_id,
+        }
+        self.save_hyperparameters(self.init_hparams)
+        self.metric = datasets.load_metric("super_glue", self.task_name, experiment_id=self.experiment_id)
```

Comment:
are these metric available in [torchmetrics](https://torchmetrics.readthedocs.io/en/stable/)?
if yes, it should use that instead to enable flexibility with DDP

Reply:
Great idea! Unfortunately, the relevant metrics are not available in torchmetrics yet but I will definitely keep an eye on it and add it later if/when they do become available.